ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/terms/LambdaFunctionTerms.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter.terms;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.execution.codegen.helpers.TriFunction;
+import io.confluent.ksql.execution.interpreter.TermEvaluationContext;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.Pair;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class LambdaFunctionTerms {
+
+  public static class LambdaFunction1Term implements Term {
```

Comment:
Could we use 

```
(Object[] args) -> { 
    ... // for loop to populate a hashmap, then
    context.pushVariableMappings(new ImmutableMap(map));
    ... 
}
``` 

to workaround the num.argument limitation? Also we would just have one term.

Reply:
I tried a few different things here, both this and `(Object... args)`.  Unfortunately, it requires both that a lambda implements some interface, and then that the parameters match it.  Since we're using three different interface on the consuming end (`Function`, `BiFunction`, `TriFunction`), we need to match their signatures.  Unfortunately, this is a limitation of that requirement and the code compiler does a similar thing.

Reply:
I cleaned this class up with a common base class and some utility methods to try to make the child classes very minimal.